### PR TITLE
Copier-host: move copier host related code to copier host specific file

### PR DESCRIPTION
**Pull request by btian1**

### CODE REVIEW

File: src/audio/copier/copier_host.c [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+#include <sof/trace/trace.h>
+#include <sof/lib/uuid.h>
+
+LOG_MODULE_REGISTER(copier_host, CONFIG_SOF_LOG_LEVEL);
+
+/* af1f5122-f399-11ed-a05b-0242ac120003 */
+DECLARE_SOF_RT_UUID("copier_host", copier_host_comp_uuid, 0xaf1f5122, 0xf399, 0x11ed,
+		    0xa0, 0x5b, 0x02, 0x42, 0xac, 0x12, 0x00, 0x03);
+
+DECLARE_TR_CTX(copier_host_comp_tr, SOF_UUID(copier_host_comp_uuid), LOG_LEVEL_INFO);
```
Comment (ranj063) [inline src/audio/copier/copier_host.c:19]: why are you declaring a new UUID @btian1 ?

File: src/include/sof/audio/host_copier.h [hunk 2]
```diff
@@ -114,5 +114,8 @@ int copier_host_create(struct comp_dev *parent_dev, struct copier_data *cd,
 		       const struct ipc4_copier_module_cfg *copier_cfg,
 		       int dir);
 void copier_host_free(struct copier_data *cd);
+int copier_host_params(struct copier_data *cd, struct comp_dev *dev,
+		       struct sof_ipc_stream_params *params);
+void copier_dma_cb(struct comp_dev *dev, size_t bytes);
```
Comment (ranj063) [inline src/include/sof/audio/host_copier.h]: stick with the same prefix please copier_host_*

File: src/include/sof/audio/host_copier.h [hunk 3]
```diff
@@ -108,5 +109,10 @@ int host_zephyr_params(struct host_data *hd, struct comp_dev *dev,
 int host_zephyr_copy(struct host_data *hd, struct comp_dev *dev, copy_callback_t cb);
 void host_update_position(struct host_data *hd, struct comp_dev *dev, uint32_t bytes);
 void host_one_shot_cb(struct host_data *hd, uint32_t bytes);
+int copier_host_create(struct comp_dev *parent_dev, struct copier_data *cd,
+		       struct comp_ipc_config *config,
+		       const struct ipc4_copier_module_cfg *copier_cfg,
+		       int dir);
+void copier_host_free(struct copier_data *cd);
```
Comment (ranj063) [inline src/include/sof/audio/host_copier.h:116]: Do these need to be in host_copier.h? or simply copier.h?

File: src/audio/copier/copier_host.c (hunk 1 above)
Comment (btian1) [reply src/audio/copier/copier_host.c:19]: Previously, I thought it is a new module, ok, seems it is part of host, let me remove new UUID.

File: src/include/sof/audio/host_copier.h (hunk 3 above)
Comment (btian1) [reply src/include/sof/audio/host_copier.h:116]: host_copier.h used to stored copier_host.c function prototype, so put it here should be fine

**Review by lyakh — APPROVED** [review]

**Comment by ranj063** [comment]
SOFCI TEST

File: src/audio/copier/copier_host.c [hunk 4]
```diff
@@ -4,6 +4,87 @@
 //
 // Author: Baofeng Tian <baofeng.tian@intel.com>
 
+#include <sof/audio/component_ext.h>
 #include <sof/trace/trace.h>
+#include <ipc4/copier.h>
+#include <sof/audio/host_copier.h>
```
Comment (RanderWang) [inline src/audio/copier/copier_host.c:10]: move "include <sof/audio/host_copier.h>" to above and follow "#include <sof/audio/component_ext.h>"

**Review by RanderWang — APPROVED** [review]
LGTM

**Review by kv2019i — APPROVED** [review]
Thanks, looks good and the naming makes more sense now,

**Comment by kv2019i** [comment]
Mandatory CI checks pass. One DUT out of business in https://sof-ci.01.org/sofpr/PR7633/build8492/devicetest/index.html , otherwise known fails. 
@ranj063 Feel free to merge if this is now ok to you.

**Review by ranj063 — APPROVED** [review]

**Pull request merged**